src/many-abci/src/module.rs

```diff
@@ -13,32 +12,44 @@ use many_types::blockchain::{
 };
 use many_types::{blockchain::RangeBlockQuery, SortOrder, Timestamp};
 use once_cell::sync::Lazy;
+use sha2::Digest;
 use std::ops::{Bound, RangeBounds};
 use tendermint::Time;
-use tendermint_rpc::{query::Query, Client};
+use tendermint_rpc::{query::Query, Client, Order};
 
 const MAXIMUM_BLOCK_COUNT: u64 = 100;
 static DEFAULT_BLOCK_LIST_QUERY: Lazy<Query> = Lazy::new(|| Query::gte("block.height", 0));
 
-fn _many_block_from_tendermint_block(block: tendermint::Block) -> Block {
+fn _many_block_from_tendermint_block<C: Client + Sync>(
+    block: tendermint::Block,
+    args: blockchain::ListArgs,
```

Comment:
I was not expecting to see this structure used here. This is the argument structure given to `list()`, and this thing is specified.